### PR TITLE
Upgrade to .NET 5

### PR DESCRIPTION
**Pull request by sondreb**
- Upgrades to .NET 5 and upgrades most of dependencies to latest versions.

### CODE REVIEW

File: src/Blockcore/Blockcore.csproj [hunk 1]
```diff
@@ -23,29 +23,29 @@
...
   <ItemGroup>
     <PackageReference Include="ConcurrentHashSet" Version="1.1.0" />
-    <PackageReference Include="DBreeze" Version="1.95.0" />
-    <PackageReference Include="LiteDB" Version="5.0.8" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning" Version="4.1.1" />
-    <PackageReference Include="Microsoft.Extensions.Http" Version="3.1.4" />
-    <PackageReference Include="NLog" Version="4.7.0" />
-    <PackageReference Include="NLog.Extensions.Logging" Version="1.6.3" />
+    <PackageReference Include="DBreeze" Version="1.97.0" />
+    <PackageReference Include="LiteDB" Version="5.0.9" />
```
Comment (dangershony) [inline src/Blockcore/Blockcore.csproj:27]: will this help the bug maybe ?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <Project>
   <PropertyGroup>
-    <Version>1.0.17</Version>
-    <RuntimeFrameworkVersion>3.1.4</RuntimeFrameworkVersion>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <Version>1.0.18</Version>
```
Comment (dangershony) [inline src/Directory.Build.props:3]: this version will only bump after the release of the nuget no?

File: src/Blockcore/Blockcore.csproj (hunk 1 above)
Comment (sondreb) [reply src/Blockcore/Blockcore.csproj:27]: No, won't fix the addressindexer.

File: src/Directory.Build.props (hunk 2 above)
Comment (sondreb) [reply src/Directory.Build.props:3]: Reverting back to .17

**Review by dangershony — APPROVED** [review]
Apart for the version comment looks good

**Review by dangershony — APPROVED** [review]

**Pull request merged**